sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -689,4 +689,38 @@ class HiveDDLSuite
       ))
     }
   }
+
+  test("datasource table property keys are not allowed") {
+    import org.apache.spark.sql.hive.HiveExternalCatalog.DATASOURCE_PREFIX
+
+    withTable("tbl") {
+      sql("CREATE TABLE tbl(a INT) STORED AS parquet")
+
+      val e = intercept[AnalysisException] {
+        sql(s"ALTER TABLE tbl SET TBLPROPERTIES ('${DATASOURCE_PREFIX}foo' = 'loser')")
+      }
+      assert(e.getMessage.contains(DATASOURCE_PREFIX + "foo"))
```

Comment:
The `HiveExternalCatalog`. My concern is that, this limitation is because of the tricks for hive metastore, so it should only exist in `HiveExternalCatalog`, the `InMemoryCatatalog` and our new metastore in the future should not have this limitation